src/RA_CodeNotes.cpp

```diff
@@ -17,35 +19,39 @@ size_t CodeNotes::Load(const std::string& sFile)
     Clear();
 
     SetCurrentDirectory(NativeStr(g_sHomeDir).c_str());
-    FILE* pf = nullptr;
-    if (fopen_s(&pf, sFile.c_str(), "rb") == 0)
+
+    std::ifstream ifile{ sFile, std::ios::binary };
+
+    Document doc;
+    IStreamWrapper isw{ ifile };
+    doc.ParseStream(isw);
+
+    if (!doc.HasParseError())
+
     {
-        Document doc;
-        doc.ParseStream(FileStream(pf));
-        if (!doc.HasParseError())
-        {
-            ASSERT(doc["CodeNotes"].IsArray());
+        ASSERT(doc["CodeNotes"].IsArray());
 
-            const Value& NoteArray = doc["CodeNotes"];
+        // use auto when you can to prevent implicit conversions (you get warnings on higher levels)
+        const auto& NoteArray = doc["CodeNotes"];
+
+        for (auto& i : NoteArray.GetArray())
+        {
+            if (i["Note"].IsNull())
+                continue;
 
-            for (SizeType i = 0; i < NoteArray.Size(); ++i)
-            {
-                const Value& NextNote = NoteArray[i];
-                if (NextNote["Note"].IsNull())
-                    continue;
+            const std::string& sNote = i["Note"].GetString();
+            if (sNote.length() < 2)
+                continue;
 
-                const std::string& sNote = NextNote["Note"].GetString();
-                if (sNote.length() < 2)
-                    continue;
+            const std::string& sAddr = i["Address"].GetString();
+            auto nAddr = static_cast<ByteAddress>(std::strtoul(sAddr.c_str(), nullptr, 16));
 
-                const std::string& sAddr = NextNote["Address"].GetString();
-                ByteAddress nAddr = static_cast<ByteAddress>(std::strtoul(sAddr.c_str(), nullptr, 16));
-                const std::string& sAuthor = NextNote["User"].GetString();	//	Author?
+            // Author?
+            const std::string& sAuthor = i["User"].GetString();
 
-                m_CodeNotes.insert(std::map<ByteAddress, CodeNoteObj>::value_type(nAddr, CodeNoteObj(sAuthor, sNote)));
-            }
+            auto code_pair{ std::make_pair(nAddr, CodeNoteObj{ sAuthor, sNote }) };
+            m_CodeNotes.insert(code_pair);
```

Comment:
TODO: Make proper move semantics for rvalue refs. Using const members with rvalues dangles the ref.